core/src/main/java/jenkins/model/GlobalCloudConfiguration.java

```diff
@@ -1,30 +1,56 @@
 package jenkins.model;
 
 import hudson.Extension;
+import hudson.RestrictedSince;
+import hudson.model.Descriptor;
+import hudson.model.RootAction;
 import hudson.slaves.Cloud;
+import hudson.util.FormApply;
 import net.sf.json.JSONObject;
 import org.jenkinsci.Symbol;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
 import org.kohsuke.stapler.StaplerRequest;
+import org.kohsuke.stapler.StaplerResponse;
+import org.kohsuke.stapler.verb.POST;
 
+import javax.annotation.CheckForNull;
+import javax.servlet.ServletException;
 import java.io.IOException;
 
 /**
- * Adds the {@link Cloud} configuration to the system config page.
+ * Provides a configuration form for {@link Jenkins#clouds}.
  *
- * <p>
- * This object just acts as a proxy to configure {@link Jenkins#clouds}
- *
- * @author Kohsuke Kawaguchi
+ * Has been overhauled in Jenkins 2.XXX to no longer contribute to Configure System, but be a standalone form.
  */
-@Extension(ordinal=-100) @Symbol("cloud") // historically this was placed at the very end of the configuration page
-public class GlobalCloudConfiguration  extends GlobalConfiguration {
+@Extension
+@Symbol("cloud")
+@Restricted(NoExternalUse.class)
+@RestrictedSince("TODO")
+public class GlobalCloudConfiguration implements RootAction {
```

Comment:
This change here seems to break the public API.

Reply:
Yes, given the lack of usages in plugins (both via GH search and `jenkins-infra/usage-in-plugins`, I decided to overhaul the class and restrict it in the process, instead of dancing around that.

Reply:
I don't think this PR will impact JCasC but if anything out of this change does it will be this line. Needs testing

Reply:
In the past when we moved for example the tool configurations out of the global config page it was done in a different way. IIRC it was still left as a `GlobalConfiguration` class but given a new category that got excluded from global config page and then included on a ManagementLink page.
Why not use a similar approach here? The old class could be left quite intact and you can add a separate RootAction if that shows the management page?

Reply:
This is slightly different since there's only one hetero-list on this page ever -- that of cloud descriptors. That's also why there's no new `ConfigurationCategory`.

So I chose to repurpose this class with some minor code changes, rather than write a new class that ends up just being a minimal wrapper view for the existing view fragment, and having the `doConfigure` endpoint that just delegates to the existing `configure`.